### PR TITLE
Include branprop nodeprop in gruptree

### PR DESCRIPTION
**Pull request by lindjoha**
Solves #219.  Includes BRANPROP trees and NODEPROP info in the GRUPTREE export

### CODE REVIEW

File: ecl2df/gruptree.py [hunk 1]
```diff
@@ -71,38 +71,40 @@ def df(
...
+    found_keywords = {key: False for key in keywords}
     for kword in deck:
         if kword.name == "DATES" or kword.name == "START" or kword.name == "TSTEP":
             # Whenever we encounter a new DATES, it means that
             # we have processed all the GRUPTREE and WELSPECS that
             # have occured since the last date, so this is the chance
             # to dump the parsed data. Also we dump the *entire* tree
             # at every date with a change, not only the newfound edges.
-            if currentedges and (found_gruptree or found_welspecs or found_grupnet):
+            if any([val for val in found_keywords.values()]):
```
Comment (berland) [inline ecl2df/gruptree.py:99]: check if pylint is complaining on this line, I wonder if the square brackets can be removed.

Comment (lindjoha) [reply ecl2df/gruptree.py:99]: I will check

Comment (lindjoha) [reply ecl2df/gruptree.py:99]: pylint didn't complain, but it works without the brackets so I'll take them away.

File: ecl2df/gruptree.py [hunk 2]
```diff
@@ -71,38 +71,46 @@ def df(
...
     # (which is a tuple of child and parent).
-    # The value of the dictionary is GRUPTREE or WELSPECS
-    currentedges: Dict[tuple, str] = dict()
-
-    grupnet_df: pd.DataFrame = pd.DataFrame()
-
-    found_gruptree = False  # Flags which will tell when a new GRUPTREE or
-    found_welspecs = False  # WELSPECS have been encountered.
-    found_grupnet = False  # GRUPNET has been encountered
+    currentedges: Dict[str, Dict[tuple, Dict[str, Any]]] = {
```
Comment (berland) [inline ecl2df/gruptree.py:80]: Should we replace `tuple` by `Tupe[something]`?

File: ecl2df/gruptree.py [hunk 3]
```diff
@@ -353,9 +430,17 @@ def gruptree_main(args) -> None:
...
                 print("Date: " + str(date.astype("M8[D]")))
-                for tree in edge_dataframe2dict(dframe[dframe["DATE"] == date]):
-                    print(tree_from_dict(tree))
+
+                for treetype in ["GRUPTREE", "BRANPROP"]:
+                    if treetype in df_date["KEYWORD"].unique():
+                        df_treetype = df_date[
+                            df_date["KEYWORD"].isin([treetype, "WELSPECS"])
+                        ]
+                        print(f"{treetype} trees:")
```
Comment (berland) [inline ecl2df/gruptree.py:441]: Add test for BRANPROP tree.

Comment (berland) [reply ecl2df/gruptree.py:441]: Consider whitespace between the pretty-printed trees, will it look good for all combinations of tree types?

Comment (lindjoha) [reply ecl2df/gruptree.py]: This is how it looks now. There is an empty line between the trees. Any suggestions?

```
Date: 2000-01-01
GRUPTREE trees:
FIELD
└── TMPL_A
    └── WELL_1

TMPL_B
└── WELL_2

BRANPROP trees:
FIELD
└── NODE_A
    └── TMPL_A
        └── WELL_1


Date: 2000-02-01
BRANPROP trees:
FIELD
└── NODE_A
    └── TMPL_A
        └── WELL_1
```

Comment (berland) [reply ecl2df/gruptree.py]: That looks good enough for me!

**Review by berland — APPROVED** [review]
👍🏻

**Pull request merged**